platform/security/sts/security-sts-attributequeryclaimshandler/src/main/java/org/codice/ddf/security/claims/attributequery/AttributeQueryClaimsHandler.java

```diff
@@ -0,0 +1,271 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.security.claims.attributequery;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.security.Principal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.StringTokenizer;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.cxf.rt.security.claims.ClaimCollection;
+import org.apache.cxf.sts.claims.ClaimsHandler;
+import org.apache.cxf.sts.claims.ClaimsParameters;
+import org.apache.cxf.sts.claims.ProcessedClaim;
+import org.apache.cxf.sts.claims.ProcessedClaimCollection;
+import org.opensaml.saml2.core.Assertion;
+import org.opensaml.saml2.core.Attribute;
+import org.opensaml.saml2.core.AttributeStatement;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.security.PropertiesLoader;
+import ddf.security.common.audit.SecurityLogger;
+import ddf.security.samlp.SimpleSign;
+
+public class AttributeQueryClaimsHandler implements ClaimsHandler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AttributeQueryClaimsHandler.class);
+
+    private String attributeMapLocation;
+
+    private List<String> supportedClaims;
+
+    private Map<String, String> attributeMap;
+
+    protected SimpleSign simpleSign;
+
+    protected String externalAttributeStoreUrl;
+
+    protected String issuer;
+
+    protected String destination;
+
+    protected static final String ERROR_RETRIEVING_ATTRIBUTES = "Error retrieving attributes from external attribute store [{}] for DN [{}]. ";
+
+    protected static final String ERROR_RETRIEVING_ATTRIBUTES_SEC_LOG = "Error retrieving attributes from external attribute store [%s] for DN [%s]. ";
+
+    public AttributeQueryClaimsHandler() {
+        LOGGER.debug("Starting AttributeQueryClaimsHandler");
+    }
+
+    @Override
+    public List<URI> getSupportedClaimTypes() {
+        LOGGER.debug("Getting supported claim types.");
+        List<URI> supportedClaimTypes = new ArrayList<>();
+        try {
+            for (String claim : supportedClaims) {
+                supportedClaimTypes.add(new URI(claim));
+            }
+        } catch (URISyntaxException e) {
+            LOGGER.error("Error getting supported claim types for external attribute store.", e);
```

Comment:
I think so. Other Claims Handlers (e.g PropertyFileClaimsHandler, RoleClaimsHandler} just log and continue. Maybe lowering the log level to warn would be a better idea.